From 1163c51852671a44ab1d51951d4da21a6b615c84 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Wed, 21 May 2025 07:23:26 +0300
Subject: [PATCH] [JSW-704] jgs: integrate jgs specs

With this patch, defined address map for jgs
using jgs specs macros also start using mem
sizes from jgs specs instead of hard coded
memory sizes.
Added arc region declerations to create any
32bit base address manually.

Change-Id: I17851f4e07701beb470fccb4ad685d3a0b84807c
---

diff --git a/fs/jgs/address_space.h b/fs/jgs/address_space.h
index 4549bd5..9f30f5a 100644
--- a/fs/jgs/address_space.h
+++ b/fs/jgs/address_space.h
@@ -11,6 +11,7 @@
  #define CFG_BAR_SIZE		0x10000000ull	/* 256MB */
  #define CFG_PHYS_BASE		0x0300007FF0000000ull
 
+ /* TODO: avoid using this hard coded macro, start using arc region macro from soc.h */
  #define HL_CFG_BASE			0x80000000
  #define CFG_ARC_REGION_BASE		0xF0000000
 
@@ -36,6 +37,7 @@
  #define COMP0_MNG_ICCM_BASE		mmCOMPUTE_0_CFG_FW_MEM_MNG_ARC_ICCM_BASE
  #define COMP0_MNG_DCCM_BASE		mmCOMPUTE_0_CFG_FW_MEM_MNG_ARC_DCCM_BASE
 
+ /* TODO: avoid using this hard coded macro, start using arc region macro from soc.h */
  #define LOCAL_ICCM_BASE			0
  #define LOCAL_DCCM_BASE			0x80000000
 
@@ -82,5 +84,47 @@
 
  #define ADDR_MASK(_mem_type)		(_mem_type ## _SIZE - 1)
 
+/*
+ * JGS start following to use macro from specs, instead of hard coding them.
+ * Or, can be used directly.
+ */
+#define IO_0_SP_SRAM_SIZE		(IO_0_CFG_SP_RAM_SECTION)
+#define IO_0_SBROM_SIZE			(IO_0_CFG_FW_MEM_PSOC_ROM_SECTION)
+#define IO_0_FWSRAM_SIZE		(IO_0_CFG_FW_MEM_FW_SRAM_SECTION)
+#define IO_0_RST_ARC_ICCM_SIZE		(IO_0_CFG_FW_MEM_RST_ARC_ICCM_SECTION)
+#define IO_0_RST_ARC_DCCM_SIZE		(IO_0_CFG_FW_MEM_RST_ARC_DCCM_SECTION)
+#define IO_0_MNG_ARC_ICCM_SIZE		(IO_0_CFG_FW_MEM_MNG_ARC_ICCM_SECTION)
+#define IO_0_MNG_ARC_DCCM_SIZE		(IO_0_CFG_FW_MEM_MNG_ARC_DCCM_SECTION)
+#define IO_0_PID_ARC_ICCM_SIZE		(IO_0_CFG_FW_MEM_PID_ARC_ICCM_SECTION)
+#define IO_0_PID_ARC_DCCM_SIZE		(IO_0_CFG_FW_MEM_PID_ARC_DCCM_SECTION)
+#define IO_0_MISC_ARC_ICCM_SIZE		(IO_0_CFG_FW_MEM_MISC_ARC_ICCM_SECTION)
+#define IO_0_MISC_ARC_DCCM_SIZE		(IO_0_CFG_FW_MEM_MISC_ARC_DCCM_SECTION)
+
+#define IO_0_RST_ARC_MEM_SIZE		(IO_0_RST_ARC_ICCM_SIZE + IO_0_RST_ARC_DCCM_SIZE)
+#define IO_0_MNG_ARC_MEM_SIZE		(IO_0_MNG_ARC_ICCM_SIZE + IO_0_MNG_ARC_DCCM_SIZE)
+#define IO_0_PID_ARC_MEM_SIZE		(IO_0_PID_ARC_ICCM_SIZE + IO_0_PID_ARC_DCCM_SIZE)
+#define IO_0_MISC_ARC_MEM_SIZE		(IO_0_MISC_ARC_ICCM_SIZE + IO_0_MISC_ARC_DCCM_SIZE)
+
+
+#define IO_0_SP_SRAM_BASE		(mmIO_0_CFG_SP_RAM_BASE) /* TODO: SOL Update */
+#define IO_0_FW_MEM_BASE		(mmIO_0_CFG_FW_MEM_BASE)
+#define IO_0_PSOC_FW_SRAM_BASE		(mmIO_0_CFG_FW_MEM_FW_SRAM_BASE)
+#define IO_0_RST_ARC_ICCM_BASE		(mmIO_0_CFG_FW_MEM_RST_ARC_ICCM_BASE)
+#define IO_0_MNG_ARC_ICCM_BASE		(mmIO_0_CFG_FW_MEM_MNG_ARC_ICCM_BASE)
+#define IO_0_PID_ARC_ICCM_BASE		(mmIO_0_CFG_FW_MEM_PID_ARC_ICCM_BASE)
+#define IO_0_MISC_ARC_ICCM_BASE		(mmIO_0_CFG_FW_MEM_MISC_ARC_ICCM_BASE)
+
+#define IO_0_SP_SRAM_BASE_FULL		(GET_FULL_ADDR(mmIO_0_CFG_SP_RAM_BASE)) /* TODO: SOL Update */
+#define IO_0_FW_MEM_BASE_FULL		(GET_FULL_ADDR(mmIO_0_CFG_FW_MEM_BASE))
+#define IO_0_PSOC_FW_SRAM_BASE_FULL	(GET_FULL_ADDR(mmIO_0_CFG_FW_MEM_FW_SRAM_BASE))
+#define IO_0_RST_ARC_ICCM_BASE_FULL	(GET_FULL_ADDR(mmIO_0_CFG_FW_MEM_RST_ARC_ICCM_BASE))
+#define IO_0_MNG_ARC_ICCM_BASE_FULL	(GET_FULL_ADDR(mmIO_0_CFG_FW_MEM_MNG_ARC_ICCM_BASE))
+#define IO_0_PID_ARC_ICCM_BASE_FULL	(GET_FULL_ADDR(mmIO_0_CFG_FW_MEM_PID_ARC_ICCM_BASE))
+#define IO_0_MISC_ARC_ICCM_BASE_FULL	(GET_FULL_ADDR(mmIO_0_CFG_FW_MEM_MISC_ARC_ICCM_BASE))
+
+
+#define CONV_64B_TO_ARC_FW_MEM(addr)	((addr & 0x3FFFFF) + mmIO_0_CFG_FW_MEM_BASE)
+
+
  #endif /* __FS1_ADDRESS_SPACE_H_ */
 
diff --git a/fs/jgs/soc.h b/fs/jgs/soc.h
new file mode 100644
index 0000000..ea9650d
--- /dev/null
+++ b/fs/jgs/soc.h
@@ -0,0 +1,57 @@
+/* SPDX-License-Identifier: MIT
+ *
+ * Copyright 2025 Intel, Ltd.
+ * All Rights Reserved.
+ *
+ */
+
+#ifndef _SOC_H_
+#define _SOC_H_
+
+#define MASK_32_BIT			0xFFFFFFFF
+#define MASK_28_BIT			0xFFFFFFF
+
+#define REGION_ICCM_IDX			0
+#define REGION_HBW_0_IDX		1
+#define REGION_HBW_1_IDX		2
+#define REGION_HBW_2_IDX		3
+#define REGION_HBW_3_IDX		4
+#define REGION_HBW_4_IDX		5
+#define REGION_HBW_5_IDX		6
+#define REGION_FW_MEM_IDX		7
+#define REGION_DCCM_IDX			8
+#define REGION_GEN_MEM_IDX		9
+#define REGION_GT_TILE_0_IDX		10
+#define REGION_GT_TILE_1_IDX		11
+#define REGION_GT_TILE_2_IDX		12
+#define REGION_GT_TILE_3_IDX		13
+#define REGION_LBW_DBG_IDX		14
+#define REGION_LBW_CFG_IDX		15
+
+#define REGION_BASE_ADDR(REG_IDX)	(REG_IDX << 28)
+
+#define ARC_REGION_MASK			(REGION_BASE_ADDR(0xF))
+#define LOCAL_ICCM_BASE			(REGION_BASE_ADDR(REGION_ICCM_IDX))
+#define LOCAL_DCCM_BASE			(REGION_BASE_ADDR(REGION_DCCM_IDX))
+#define FW_MEM_REG_BASE			(REGION_BASE_ADDR(REGION_FW_MEM_IDX))
+#define GEN_MEM_REG_BASE		(REGION_BASE_ADDR(REGION_GEN_MEM_IDX))
+#define CFG_SPACE_REG_BASE		(REGION_BASE_ADDR(REGION_LBW_CFG_IDX))
+
+#define FULL_ADDR_MSB			0x300007F
+
+#define CFG_MEM_FULL_BASE		0x300007ff0000000ull
+#define GEN_MEM_FULL_BASE		0x300007f00000000ull
+
+#define GET_FULL_ADDR(REG_OFF)		((REG_OFF & MASK_28_BIT) + CFG_MEM_FULL_BASE)
+
+/* This 2 macro's to convert ARC based 32B address to LBW MEM map address of 32B */
+#define CONV_ARC_REG_TO_GEN_MEM(addr)	(((addr) & MASK_28_BIT) + ((GEN_MEM_FULL_BASE) & MASK_32_BIT))
+#define CONV_ARC_REG_TO_CFG_MEM(addr)	(((addr) & MASK_28_BIT) + ((CFG_MEM_FULL_BASE) & MASK_32_BIT))
+
+#define FLASH_VALID_MASK		(SZ_2M - 1)
+#define HL_FLASH_OFFSET			0xF000000
+#define HL_FLASH_SIZE			(16 * SIZE_MB)
+#define SPI_FLASH_BASE			(GEN_MEM_REG_BASE + HL_FLASH_OFFSET)
+
+#endif /* _SOC_H_ */
+
